src/ray/gcs/redis_module/ray_redis_module.cc

```diff
@@ -511,13 +520,79 @@ int SetRemove_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int ar
     return REDISMODULE_ERR;
   }
   if (changed) {
-    return Set_DoPublish(ctx, argv, /*is_add=*/false);
+    return Set_DoPublish(ctx, argv, /*is_add=*/false, /*reply_if_ok=*/true);
   } else {
     RAY_LOG(ERROR) << "The entry to remove doesn't exist.";
   }
   return RedisModule_ReplyWithSimpleString(ctx, "OK");
 }
 
+/// Replace an entry from the set stored at a key with another entry.
+/// It publishes two notifications about the update to all subscribers, if a pubsub
+/// channel is provided and the replace operation succeeds. If the target removing
+/// entry does not exist, this command returns a failure message, otherwise, OK.
+///
+/// This is called from a client with the command:
+//
+///    RAY.SET_REPLACE <table_prefix> <pubsub_channel> <id> <data>
+///
+/// \param table_prefix The prefix string for keys in this table.
+/// \param pubsub_channel The pubsub channel name that notifications for this
+/// key should be published to. When publishing to a specific client, the
+/// channel name should be <pubsub_channel>:<client_id>.
+/// \param id The ID of the key to remove from.
+/// \param data The data contains the data to remove and data to add.
+/// \return OK if the remove succeeds, or failure message if the operation fails.
+int SetReplace_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  if (argc != 5) {
+    return RedisModule_WrongArity(ctx);
+  }
+  // Copy the arguements .
+  RedisModuleString *data = argv[4];
+  std::vector<RedisModuleString *> argv_vector(argv, argv + argc);
+  RedisModuleString **new_argv = &argv_vector[0];
+  // Split the data into 2 parts.
+  size_t data_size = 0;
+  const uint8_t *data_string =
+      reinterpret_cast<const uint8_t *>(RedisModule_StringPtrLen(data, &data_size));
+  auto root = flatbuffers::GetRoot<SetReplaceEntryData>(data_string);
+  // Remove the old data.
+  auto old_data =
+      RedisModule_CreateString(ctx, root->old_data()->data(), root->old_data()->size());
+  new_argv[4] = old_data;
+  bool changed = false;
+  bool is_add = false;
+  bool do_reply = false;
+  if (Set_DoWrite(ctx, new_argv, argc, is_add, &changed) != REDISMODULE_OK) {
```

Comment:
I think it is safer to not allocate a new variable called `new_argv` and instead pass in `argv_vector.data()` here. Also, I would probably then rename `argv_vector` to `new_argv` or something.

Reply:
Good.